arch/x86/soc/ia32/soc.c

```diff
@@ -17,3 +17,18 @@
 #include <uart.h>
 #include <device.h>
 #include <init.h>
+#include <mmustructs.h>
+
+#ifdef CONFIG_X86_MMU
+
+#ifdef CONFIG_XIP
+MMU_BOOT_REGION(CONFIG_PHYS_LOAD_ADDR, CONFIG_ROM_SIZE*1024, MMU_ENTRY_READ);
+
+MMU_BOOT_REGION(CONFIG_PHYS_RAM_ADDR, CONFIG_RAM_SIZE*1024, MMU_ENTRY_WRITE);
+#else
+MMU_BOOT_REGION(CONFIG_PHYS_LOAD_ADDR, CONFIG_RAM_SIZE*1024, MMU_ENTRY_WRITE);
+MMU_BOOT_REGION(CONFIG_LOAPIC_BASE_ADDRESS, 4096*1024, MMU_ENTRY_WRITE);
```

Comment:
You need to move these APIC definitions outside of the CONFIG_XIP block, they apply in either case.
You also need a memory range for the HPET timer driver if I'm not mistaken.